indico/core/db/sqlalchemy/protection.py

```diff
@@ -521,14 +521,16 @@ def update_principal(self, principal, read_access=None, full_access=None, permis
                                        old_data=old_data, quiet=quiet)
         return entry
 
-    def get_manager_list(self, recursive=False, include_groups=True):
+    def get_manager_list(self, recursive=False, include_groups=True, permission=None, explicit=False):
         managers = {
             x.principal
             for x in self.acl_entries
-            if x.has_management_permission() and (include_groups or x.type == PrincipalType.user)
+            if (x.has_management_permission(permission=permission, explicit=explicit) and
+                (include_groups or x.type == PrincipalType.user))
         }
         if recursive and self.protection_parent:
-            managers.update(self.protection_parent.get_manager_list(recursive=True, include_groups=include_groups))
+            managers.update(self.protection_parent.get_manager_list(recursive=True, include_groups=include_groups,
+                                                                    permission=permission, explicit=explicit))
```

Comment:
That's almost never going to be correct. Specific permissions are not inherited, so I'd either disallow calling this method with `recursive and (permission or explicit)` or omit this here.